nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/service/RenditionServiceImpl.java

```diff
@@ -535,28 +528,44 @@ private void removeStoredRenditions(Map<String, List<String>> liveDocumentRefsTo
 
     @Override
     public Rendition getDefaultRendition(DocumentModel doc, String reason, Map<String, Serializable> extendedInfos) {
-        if (defaultRenditionDescriptor == null) {
+        return getDefaultRendition(doc, reason, false, extendedInfos);
+    }
+
+    @Override
+    public Rendition getDefaultRendition(DocumentModel doc, String reason, boolean store, Map<String, Serializable> extendedInfos) {
+        DefaultRenditionDescriptor resolvedDesc = null;
+        for (int i = defaultRenditionDescriptors.size() - 1; i >= 0; i--) {
+            DefaultRenditionDescriptor d = defaultRenditionDescriptors.get(i);
+            if ((StringUtils.isEmpty(reason) && StringUtils.isEmpty(d.reason))
+                    || (reason != null && reason.equals(d.reason))) {
+                resolvedDesc = d;
+                break;
+            }
+        }
+        if (resolvedDesc == null) {
```

Comment:
Wouldn't it make more sense to retrieve all rendition descriptors with the given reason and evaluate until != null? This would make it easier to set up a custom default rendition for new document types.

Reply:
I am not sure, this will defeat the ability to simply override an existing contribution

Reply:
Yes, we'd need to have another key besides reason, but without it I'm not sure how addons will be able to contribute new doc types with dowload or publish renditions...

Reply:
you get a point